controllers/registry.go

```diff
@@ -251,25 +251,58 @@ func (r *DPAReconciler) getSecretNameAndKeyforBackupLocation(bslspec oadpv1alpha
 		}
 	}
 	if bslspec.Velero != nil {
-		return r.getSecretNameAndKey(bslspec.Velero, oadpv1alpha1.DefaultPlugin(bslspec.Velero.Provider))
+		return r.getSecretNameAndKey(bslspec.Velero.Config, bslspec.Velero.Credential, oadpv1alpha1.DefaultPlugin(bslspec.Velero.Provider))
 	}
 
 	return "", ""
 }
 
-func (r *DPAReconciler) getSecretNameAndKey(bslSpec *velerov1.BackupStorageLocationSpec, plugin oadpv1alpha1.DefaultPlugin) (string, string) {
+func (r *DPAReconciler) getSecretNameAndKeyforSnapshotLocation(vslspec oadpv1alpha1.SnapshotLocation) (string, string) {
+
+	if vslspec.Velero != nil {
+		return r.getSecretNameAndKey(vslspec.Velero.Config, vslspec.Velero.Credential, oadpv1alpha1.DefaultPlugin(vslspec.Velero.Provider))
+	}
+
+	return "", ""
+}
+
+func (r *DPAReconciler) getSecretNameAndKey(config map[string]string, credential *corev1.SecretKeySelector, plugin oadpv1alpha1.DefaultPlugin) (string, string) {
+	// Assume default values unless user has overriden them
+	secretName := credentials.PluginSpecificFields[plugin].SecretName
+	secretKey := credentials.PluginSpecificFields[plugin].PluginSecretKey
+	if _, ok := config["credentialsFile"]; ok {
+		if secretName, secretKey, err :=
+			credentials.GetSecretNameKeyFromCredentialsFileConfigString(config["credentialsFile"]); err == nil {
+			r.Log.Info(fmt.Sprintf("credentialsFile secret: %s, key: %s", secretName, secretKey))
+			return secretName, secretKey
+		}
+	}
+	// check if user specified the Credential Name and Key
+	if credential != nil {
+		if len(credential.Name) > 0 {
+			secretName = credential.Name
+		}
+		if len(credential.Key) > 0 {
+			secretKey = credential.Key
+		}
+	}
+
+	return secretName, secretKey
+}
+
+func (r *DPAReconciler) getSecretNameAndKeyFromSnapshotLocation(vslspec *velerov1.VolumeSnapshotLocationSpec, plugin oadpv1alpha1.DefaultPlugin) (string, string) {
```

Comment:
Where is this func used?

Reply:
oops, nowhere, not needed anymore, common code is being used,  removed it. Thanks!